### PR TITLE
[WIP] Generate search.json file

### PR DESCRIPTION
**Pull request by khalidabuhakmeh**
This could eventually be pushed into ElasticSearch and provide a better search capability.

### CODE REVIEW

**Comment by khalidabuhakmeh** [comment]
need to add an `id` field, which will most likely be a slug of the url

**Comment by kendaleiv** [comment]
Who/what is the intended user of this?

**Comment by khalidabuhakmeh** [comment]
This is the first step in seeing if a Jekyll site is easily indexable in ElasticSearch.

**Comment by khalidabuhakmeh** [comment]
I imagine an Azure Webjob or something similar would retrieve this file and then push it into an ElasticSearch provider.

**Comment by kendaleiv** [comment]
That's not a bad plan, provided we are OK with a *pull* solution. For our current purposes, I'll assume we are.

--

Alternatively, if we want a push solution: http://allizad.com/2016/05/06/elasticserch-for-jekyll/index.html https://github.com/omc/searchyll

That solution does complicate matters if hosting on GitHub pages with PR workflow, though. In that case, we'd probably want the CI `master` build to handle the updating. We may even choose that path for non-GitHub pages hosted items, if we went that route.

**Comment by khalidabuhakmeh** [comment]
image

**Comment by khalidabuhakmeh** [comment]
So the import works pretty quickly there is just a few things to note:

1. `localhost:9200` is hardcoded, not ideal
2. there is not mapping which improves the search right now
3. the index name is hardcoded to `jekyll`

As a side note, naming an index `content` doesn't seem to work.

File: package.json [hunk 1]
```diff
@@ -23,6 +23,7 @@
   "license": "UNLICENSED",
   "devDependencies": {
     "markdown-proofing": "latest",
-    "shelljs": "^0.7.0"
+    "shelljs": "^0.7.0",
+    "elasticsearch": "^11.0.1"
```
Comment (khalidabuhakmeh) [inline package.json]: I'm not a `100%` sure if this should be a dev dependency. @kendaleiv or @comfroels will tell me.

**Comment by khalidabuhakmeh** [comment]
Using `Sense` Chrome Extension

```text
POST jekyll/_search 
{
    "query": {
       "match": {
          "content": "OpenID"
       }
       
    }
}
```

File: package.json (hunk 1 above)
Comment (kendaleiv) [inline package.json]: Hmm, it's not required to run the site. Perhaps the way to think about it is if `npm install --production` was used, would you want it to be installed or not?

Comment (comfroels) [inline package.json]: Yeah, I'd say it's a dev dependency

**Pull request closed without merging**